dali/operators/decoder/audio/audio_decoder_impl.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/decoder/audio/audio_decoder_impl.h"
+#include "dali/kernels/signal/downmixing.h"
+
+namespace dali {
+
+template <typename T>
+span<char> as_raw_span(T *buffer, ptrdiff_t length) {
```

Comment:
How much it overlaps with audio_decoder_op.cc?

Reply:
The idea is that I extracted the audio decoder implementation details here so that I can reuse them. I plan to modify the audio decoder to use this but I decided to leave this for later. Any thoughts?

Reply:
I have a mixed filing about pushing halfway done things like this. I would at least extract this part and reuse. You can leave the rest untouched.

Reply:
I modified the audio decoder op now to reuse this implementation